src/core/ext/filters/channel_idle/channel_idle_filter.cc

```diff
@@ -111,12 +113,21 @@ struct MaxAgeFilter::Config {
             .value_or(kDefaultMaxConnectionAgeGrace);
     // generate a random number between 1 - kMaxConnectionAgeJitter and
     // 1 + kMaxConnectionAgeJitter
-    const double multiplier =
-        rand() * kMaxConnectionAgeJitter * 2.0 / RAND_MAX + 1.0 -
-        kMaxConnectionAgeJitter;
+    struct BitGen {
+      Mutex mu;
+      absl::BitGen bit_gen ABSL_GUARDED_BY(mu);
+      double MakeUniformDouble(double min, double max) {
+        MutexLock lock(&mu);
+        return absl::Uniform(bit_gen, min, max);
+      }
+    };
+    static NoDestruct<PerCpu<BitGen>> bit_gen(PerCpuOptions().SetMaxShards(8));
```

Comment:
Would it be better to just instantiate a new `absl::BitGen` object each time we call this?  I believe this code runs at connection establishment time, not per-call, right?  If so, it shouldn't be that performance sensitive and therefore may not be worth the complexity of `PerCpu<>` and synchronization.  Also, it seems like allocating 8 `BitGen` objects will increase memory usage, which may also not be the right trade-off.

Reply:
Connection establishment needs significant optimization, so I'm not prepared to just not worry here, and constructing a BitGen just to generate one number is about 20x more expensive than using a shared one - and about 10x more expensive than using a shared one under a mutex.

We should probably look into a way of doing a shared RNG more globally.